docker/Dockerfile.amd

```diff
@@ -2,94 +2,170 @@
 # This Dockerfile contains AMD compilers
 ##############################################################
```

Comment:
Perhaps we add a reference to https://github.com/amd/InfinityHub-CI/tree/main/base-gpu-mpi-rocm-docker ?